libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -178,25 +208,26 @@ internal class RouteAlternativesController constructor(
         nativeAlternatives: List<RouteAlternative>,
         block: (List<NavigationRoute>, RouterOrigin) -> Unit,
     ) = mainJobControl.scope.launch {
-        val alternatives = nativeAlternatives.mapIndexedNotNull { index, routeAlternative ->
-            val expected = parseNativeDirectionsAlternative(
-                ThreadController.IODispatcher,
-                routeAlternative
-            )
-            if (expected.isValue) {
-                expected.value
-            } else {
-                logE(
-                    """
-                            |unable to parse alternative at index $index;
-                            |failure for response: ${routeAlternative.route.responseJson}
-                        """.trimMargin(),
-                    LOG_CATEGORY
+        val alternatives: List<NavigationRoute> =
+            nativeAlternatives.mapIndexedNotNull { index, routeAlternative ->
+                val expected = parseNativeDirectionsAlternative(
+                    ThreadController.IODispatcher,
```

Comment:
Out of scope of this PR: 
as we're serialising JSON from a string in RAM, maybe it's more about computation and we should use computation thread pool `Dispathers.Default`?